addons/binding/org.openhab.binding.cups/src/main/java/org/openhab/binding/cups/handler/CupsHandler.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cups.handler;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.cups4j.CupsPrinter;
+import org.cups4j.WhichJobsEnum;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.cups.CupsBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CupsHandler} is responsible for handling commands, which are sent
+ * to one of the channels.
+ * 
+ * @author Tobias Braeutigam - Initial contribution
+ */
+public class CupsHandler extends BaseThingHandler implements
+		PrinterStatusListener {
+
+	private Logger logger = LoggerFactory.getLogger(CupsHandler.class);
+
+	private URL url;
+	private String name;
+	
+	private int refresh = 60; // refresh every minute as default
+	ScheduledFuture<?> refreshJob;
+	
+	private CupsBridgeHandler bridgeHandler;
+
+	public CupsHandler(Thing thing) {
+		super(thing);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void initialize() {
+		Configuration config = getThing().getConfiguration();
+		try {
+			Object obj = config.get(CupsBindingConstants.PRINTER_PARAMETER_URL);
+			if (obj instanceof URL)
+				url = (URL) obj;
+			else if (obj instanceof String)
+				url = new URL((String)obj) ;
+		} catch (MalformedURLException e) {
+			logger.error("malformed url {}, printer thing creation failed",config.get(CupsBindingConstants.PRINTER_PARAMETER_URL));
+		}
+		name = (String) config.get(CupsBindingConstants.PRINTER_PARAMETER_NAME);
+		// until we get an update put the Thing offline
+		updateStatus(ThingStatus.OFFLINE);
+		deviceOnlineWatchdog();
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.smarthome.core.thing.binding.BaseThingHandler#dispose()
+	 */
+	@Override
+	public void dispose() {
+		if (refreshJob != null && !refreshJob.isCancelled()) {
+			refreshJob.cancel(true);
+			refreshJob = null;
+		}
+		updateStatus(ThingStatus.OFFLINE);
```

Comment:
you do not need (or rather must not) update the status here anymore - the framework takes care of this.